drivers/vulkan/rendering_device_driver_vulkan.cpp

```diff
@@ -3470,6 +3470,7 @@ Vector<uint8_t> RenderingDeviceDriverVulkan::shader_compile_binary_from_spirv(Ve
 		offset += sizeof(uint32_t);
 		encode_uint32(sizeof(ShaderBinary::Data), binptr + offset);
 		offset += sizeof(uint32_t);
+		offset += sizeof(uint32_t); // Pad to align ShaderBinary::Data to 8 bytes.
```

Comment:
To make sure we don't get regressions from `ShaderBinary::Data` changing size in the future, should we have some `static_assert` for the assumption we're making here / above / below?

Just mentioning this to flag it, maybe wait for @clayjohn's input before doing it, as I don't have the full picture of how all this stuff is aligned.

Reply:
We could also just turn it into a real struct perhaps?

Reply:
I'm also not entirely sure why the size is in there at all since it seems that sizeof(ShaderBinary::Data) is constant.

Reply:
I don't have any strong opinions. CC @RandomShaper who knows this code better than I do

Reply:
We could at least add some getters and setters to a structure with a flexible member so that the framing of the packet doesn't have to implemented twice in two different places? One where the packet gets created, and then again where it gets read?

I don't think that should stop us from merging this in my opinion. This fixes a bug, but leaves the hard to maintain code in place. If desired I can propose a PR to make it easier to maintain separately?

Reply:
> To make sure we don't get regressions from `ShaderBinary::Data` changing size in the future, should we have some `static_assert` for the assumption we're making here / above / below?

The actual size of `ShaderBinary::Data` is dynamically determined. And it changing size wouldn't change alignment of it. The problem was that we had it 12 bytes into the structure, but the structure has `uint64_t vertex_input_mask = 0;`. This means that the whole structure needs to be aligned to 8 bytes, not 4. That is what the extra padding is for.

Reply:
Hi everyone! Regarding the changing of `sizeof(ShaderBinary::Data)` see [my reply](https://github.com/godotengine/godot/pull/100325#issuecomment-2540095483).

A static_assert would indeed make things safer, but ultimately the responsibility of guarding that is in the function being modified by this PR, because it's a **_versioned_** binary data. The PR forgot to bump `ShaderBinary::VERSION`.

Of course a static_assert would add an extra layer of reminders "hey! remember to bump `ShaderBinary::VERSION`!".

Reply:
I really don't know what static assert would have caught this, or the version. The size of `ShaderBinary::Data` did not change. I added some extra padding to an entirely "free form" memory segment without any structure to it.

Reply:
@akien-mga yeah, I get it. But due to the way this code works now I don't think there's any way to check it from the "outside" without some kind of unit test. I think it might be better to use packed structs for this and then we could in fact use some assertions to test this potentially.